src/lib.rs

```diff
@@ -727,18 +734,22 @@ fn set(
 #[cfg(feature = "failpoints")]
 macro_rules! fail_point {
     ($name:expr) => {{
-        $crate::eval($name, |_| {
-            panic!("Return is not supported for the fail point \"{}\"", $name);
-        });
+        $crate::eval(
+            $name,
+            |_| {
+                panic!("Return is not supported for the fail point \"{}\"", $name);
+            },
+            || true,
+        );
     }};
     ($name:expr, $e:expr) => {{
-        if let Some(res) = $crate::eval($name, $e) {
+        if let Some(res) = $crate::eval($name, $e, || true) {
             return res;
         }
     }};
     ($name:expr, $cond:expr, $e:expr) => {{
-        if $cond {
-            fail_point!($name, $e);
+        if let Some(res) = $crate::eval($name, $e, || $cond) {
```

Comment:
This makes it hard to share mutable states between $cond and `$e`.

Reply:
If true then we better add a test case that captures the expected behavior.

@gengliqi can you try adding a test case that shares mutable references between the two expressions and seeing if they worked previously but don't work after this patch?